components/bms_worker/src/Cooling.c

```diff
@@ -0,0 +1,156 @@
+
+/**< Module header */
+#include "Cooling.h"
+
+/**< Driver Includes */
+#include "HW_Fans.h"
+
+/**< Other Includes */
+#include "Environment.h"
+#include "Module.h"
+
+/******************************************************************************
+ *                              D E F I N E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                              E X T E R N S
+ ******************************************************************************/
+
+extern Environment_S ENV;
+
+/******************************************************************************
+ *                             T Y P E D E F S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                               M A C R O S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                           P U B L I C  V A R S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                         P R I V A T E  V A R S
+ ******************************************************************************/
+
+/**
+ * @brief  Stores the public Cooling Manager struct
+ */
+Cooling_Mngr_S COOLING;
+
+
+/******************************************************************************
+ *            P U B L I C  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *          P R I V A T E  F U N C T I O N  P R O T O T Y P E S
+ ******************************************************************************/
+
+/******************************************************************************
+ *                       P U B L I C  F U N C T I O N S
+ ******************************************************************************/
+
+/**
+ * @brief  Cooling Module Init function
+ */
+static void Cooling_Init()
+{
+    COOLING.state      = COOLING_INIT;
+    COOLING.percentage = 0;
+    FANS_Init();
+
+    COOLING.state = COOLING_INIT;
+}
+
+/**
+ * @brief  Cooling Module 1Hz periodic function
+ */
+static void Cooling1Hz_PRD(void)
+{
+    static uint8_t step = 0;
+
+    switch (COOLING.state)
+    {
+        case COOLING_INIT:
+            step += 20;
+            COOLING.percentage = (step <= 100) ? step : 200 - step;
+
+            if (step >= 200)
+                COOLING.state = COOLING_OFF;
+            break;
+        case COOLING_ON:
+        case COOLING_OFF:
+            if (ENV.values.cells.max_temp < 350)
```

Comment:
should do a simple linear interpolation instead of this. Or maybe a define a mathematical function that translates temp to percentage.